### PR TITLE
feat: support code scan with sonarqube

### PR DESCRIPTION
**Pull request by zhujian7**
**What this PR does / why we need it**:

Add your description

**Which issue(s) this PR is related to** *(optional, link to 3rd issue(s))*:

Fixes #

Reference to #

**Special notes for your reviewer**:

/cc @ladjzero @cd1989 @supereagle @qiuxiaolian 

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: docs/api/v1/api.md [hunk 1]
```diff
@@ -1678,4 +1707,134 @@ Success:
...
+```
+{
+    "name": "sonar1",  // can not update
+    "alias": "alias",
+    "type": "SonarQube",  // can not update
+    "sonarqube": {
+        "description": "This is my first sonar qube instance.",
+        "address": "http://192.168.21.100:9000",
+        "token": "f399878566d5d6a3de1759222a4b5eb15cac51de",
+        "user":"" // Optional, stored for front-end to display.
```
Comment (ladjzero) [inline docs/api/v1/api.md]: 需要展示“创建时间”

Comment (zhujian7) [reply docs/api/v1/api.md]: have fixed

File: docs/api/v1/api.md [hunk 2]
```diff
@@ -175,6 +184,19 @@
 | API | Path | Detail |
 | --- | --- | --- |
 | List | GET `/api/v1/configtemplates` | WIP, [link](#list-config-templates) |
+
+### Integration API
+- [IntegrationObject](#integrationobject)
+
+| API | Path | Detail |
+| --- | --- | --- |
+| Create | GET `/api/v1/integrations` | WIP, [link](#create-integration) |
```
Comment (cd1989) [inline docs/api/v1/api.md]: Why still leave `WIP` ?

Comment (zhujian7) [reply docs/api/v1/api.md]: lol, my bad.

File: pkg/api/v1/descriptor/descriptor.go [hunk 3]
```diff
@@ -51,16 +52,29 @@ func newLogMiddleware() definition.Middleware {
...
+				time.Now().Format("02/Jan/2006:15:04:05 -0700"),
+				req.Method,
+				req.URL.RequestURI(),
+				req.Proto,
+				resp.StatusCode,
+				resp.ContentLength,
+				time.Since(startTime),
+			)
+		} else {
+			//log.V(log.LevelDebug).Infof("%s - [%s] \"%s %s %s\" %d %d %v",
```
Comment (supereagle) [inline pkg/api/v1/descriptor/descriptor.go]: Can remove if not needed

File: pkg/worker/integrate/manager.go [hunk 4]
```diff
@@ -0,0 +1,77 @@
+package integrate
+
+import (
+	"fmt"
+
+	"github.com/caicloud/cyclone/pkg/api"
+)
+
+// ITGProvider is an interface for ingetration.
+type ITGProvider interface {
```
Comment (supereagle) [inline pkg/worker/integrate/manager.go]: `ITG` is a little confused, use full word?

Comment (zhujian7) [reply pkg/worker/integrate/manager.go]: Full word is a little long.

**Pull request merged**